rai/client.go

```diff
@@ -885,108 +885,123 @@ func (c *Client) ListOAuthClients() ([]OAuthClient, error) {
 
 func (c *Client) DeleteModel(
 	database, engine, name string,
-) (*TransactionResult, error) {
+) (*TransactionAsyncResult, error) {
 	return c.DeleteModels(database, engine, []string{name})
 }
 
+func (c *Client) DeleteModelAsync(
+	database, engine, name string,
+) (*TransactionAsyncResult, error) {
+	return c.DeleteModelsAsync(database, engine, []string{name})
+}
+
 func (c *Client) DeleteModels(
 	database, engine string, models []string,
-) (*TransactionResult, error) {
-	var result TransactionResult
-	tx := Transaction{
-		Region:   c.Region,
-		Database: database,
-		Engine:   engine,
-		Mode:     "OPEN",
-		Readonly: false}
-	data := tx.Payload(makeDeleteModelsAction(models))
-	err := c.Post(PathTransaction, tx.QueryArgs(), data, &result)
-	if err != nil {
-		return nil, err
+) (*TransactionAsyncResult, error) {
+	queries := make([]string, 0)
+	for _, model := range models {
+		queries = append(queries, fmt.Sprintf("def delete:rel:catalog:model[\"%s\"] = rel:catalog:model[\"%s\"]", model, model))
 	}
-	return &result, err
+
+	return c.Execute(database, engine, strings.Join(queries, "\n"), nil, false)
+}
+
+func (c *Client) DeleteModelsAsync(
+	database, engine string, models []string,
+) (*TransactionAsyncResult, error) {
+	queries := make([]string, 0)
+	for _, model := range models {
+		queries = append(queries, fmt.Sprintf("def delete:rel:catalog:model[\"%s\"] = rel:catalog:model[\"%s\"]", model, model))
+	}
+
+	return c.ExecuteAsync(database, engine, strings.Join(queries, "\n"), nil, false)
 }
 
 func (c *Client) GetModel(database, engine, model string) (*Model, error) {
-	var result listModelsResponse
-	tx := NewTransaction(c.Region, database, engine, "OPEN")
-	data := tx.Payload(makeListModelsAction())
-	err := c.Post(PathTransaction, tx.QueryArgs(), data, &result)
+	models, err := c.ListModels(database, engine)
```

Comment:
Why would we query out all models just to return a single one? We should issue a separate query here targeting a single model. This is no different than if a database has 1,000,000 products in it. You wouldn't query out all 1,000,000 just to return a single one.

Reply:
that's right :grimacing: will fix this

Reply:
this was done previously as it was the only way to retrieve a single model, with the expectation that we would eventually replace with an appropriate API call.